tests/schema/test_interface.py

```diff
@@ -107,6 +107,10 @@ class Entity:
     class Anime(Entity):
         name: str
 
+        @classmethod
+        def is_type_of(cls, obj, _info) -> bool:
+            return isinstance(obj, AnimeORM)
```

Comment:
is it required now? 🤔

Reply:
Yeah, with this change you'd have to explicitly resolve the type for non-strawberry types 🙁

I wasn't able to figure out how to support a way that duck typing works implicitly.

Having any `is_type_of` set means that [this branch](https://github.com/graphql-python/graphql-core/blob/f145dc9d0ceb6943b8adcbf86ece3c6403b34bb3/src/graphql/execution/execute.py#L915-L935) is taken within `graphql-core` while executing. It does extra validation on the return type and would now error, since the implicit `is_type_of` definition only works with strawberry types.

Previously, `is_type_of` was `None`. If we change the implicit `is_type_of` definition to `None` instead of checking against the strawberry type, the duck typing test passes (since `Anime` is an object type and not an interface) but the above tests with fields that return interfaces begin to fail (since they can't be resolved to object types anymore).

Supporting both usage patterns I think would mean only defining `resolve_type` on the interface if none of the implementing types have an `is_type_of` defined. That's kind of hard though, since the object types are defined after the interface and I didn't see a good way to un-define `resolve_type` after the fact. Having a callable there at all changes the behavior and makes it ignore all the `is_type_of` definitions on the object type.

Reply:
thanks for the in-depth explanation :) I'll think about this in the next few days and see if we can find a solution, otherwise this becomes a big breaking change

I guess we can automatically provide a `is_type_of` for interfaces maybe (if none is defined)

Reply:
There is one that's automatically defined to resolve strawberry types correctly, it just doesn't handle duck types:

https://github.com/strawberry-graphql/strawberry/pull/1406/files#diff-9db4ecb0b6a9104731da4af6d847961cbc88e4085f4f23f4fcfceb1b22e2fd35R287-R288

```python
def is_type_of(obj: Any, _info: GraphQLResolveInfo) -> bool:
    return isinstance(obj, object_type.origin)
```

Currently the other test cases in this file pass without an explicit classmethod, since they all only use strawberry types.

How would you determine whether a non-strawberry object matches the strawberry type without any extra information from the implementer?

Reply:
> How would you determine whether a non-strawberry object matches the strawberry type without any extra information from the implementer? 

I think I did misunderstand your comment then! I thought it was always required!

Cool! I'll take a final look tonight and I'll merge ^^